string.go

```diff
@@ -747,6 +748,76 @@ func (s *String) MatchAll(re string) []Match {
 	return ret
 }
 
+// IsASCII succeeds if all string characters are ASCII
+//
+// Example:
+//
+// str := NewString(t, "Hello")
+// str.IsASCII()
+func (s *String) IsASCII() *String {
+	s.chain.enter("IsASCII()")
+	defer s.chain.leave()
+
+	if s.chain.failed() {
+		return s
+	}
+
+	isASCII := true
+	for _, c := range s.value {
+		if c > unicode.MaxASCII {
+			isASCII = false
+			break
+		}
+	}
+
+	if !isASCII {
+		s.chain.fail(AssertionFailure{
+			Type:   AssertContainsSubset,
+			Actual: &AssertionValue{s.value},
+			Errors: []error{
+				errors.New("expected: all string characters are ascii"),
+			},
+		})
+	}
+
+	return s
+}
+
+// IsNotASCII succeeds if all string characters are not ASCII
+//
+// Example:
+//
+// str := NewString(t, "こんにちは")
+// str.IsNotASCII()
+func (s *String) IsNotASCII() *String {
+	s.chain.enter("IsNotASCII()")
+	defer s.chain.leave()
+
+	if s.chain.failed() {
+		return s
+	}
+
+	isASCII := false
+	for _, c := range s.value {
+		if c <= unicode.MaxASCII {
+			isASCII = true
+			break
+		}
+	}
```

Comment:
Here we check that all characters are non-ASCII.

However, in practice we would usually call a string non-ASCII if it has at least one non-ASCII character. For example, we would call "Straße" non-ASCII string. Also, logical negation of IsASCII() would give us the same result.

So I suggest to fix the algorithm, error message, and comment.

Reply:
I fix algorithm ed4c868